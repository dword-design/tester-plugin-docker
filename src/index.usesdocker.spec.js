import { endent } from '@dword-design/functions'
import tester from '@dword-design/tester'
import packageName from 'depcheck-package-name'
import execa from 'execa'
import outputFiles from 'output-files'
import withLocalTmpDir from 'with-local-tmp-dir'

export default tester(
  {
    works: async () => {
      await outputFiles({
        Dockerfile: endent`
          FROM node:12-alpine
          WORKDIR /app
        `,
        'index.spec.js': endent`
          import tester from '@dword-design/tester'
          import execa from 'execa'
          import { endent, property } from '@dword-design/functions'
          import self from '../src'

          export default tester({
            works: async () => expect(
              execa('docker', ['run', '--rm', '-v', \`\${process.cwd()}:/app\`, 'self', 'ls'], { all: true })
                |> await
                |> property('all')
            )
              .toEqual(endent\`
                Dockerfile
                index.spec.js
              \`)
          }, [self()])
        `,
      })
      await execa.command(
        `mocha --ui ${packageName`mocha-ui-exports-auto-describe`} --timeout 10000 index.spec.js`,
        { stderr: 'inherit' }
      )
    },
  },
  [
    {
      transform: test =>
        function () {
          return withLocalTmpDir(() => test.call(this))
        },
    },
  ]
)
