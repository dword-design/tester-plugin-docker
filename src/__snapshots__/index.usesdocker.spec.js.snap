// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.usesdocker works 1`] = `
"

  index
    âœ“ works


  1 passing

------------------|---------|----------|---------|---------|-------------------
File              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
------------------|---------|----------|---------|---------|-------------------
All files         |       0 |      100 |       0 |       0 |                   
 index.dockerfile |       0 |      100 |     100 |       0 | 0                 
 index.spec.js    |       0 |      100 |       0 |       0 | 1-7               
------------------|---------|----------|---------|---------|-------------------"
`;
